pkg/internal/token/options.go

```diff
@@ -63,6 +63,9 @@ func GetSupportedLogins() string {
 }
 
 func NewOptions() Options {
+	if env_TokenCacheDir := os.Getenv("KUBECACHEDIR"); env_TokenCacheDir != "" {
```

Comment:
let's not change the global variable. how about this?

```go
func NewOptions() Options {
	tokenCacheDir := DefaultTokenCacheDir
	if envTokenCacheDir := os.Getenv("KUBECACHEDIR"); envTokenCacheDir != "" {
		tokenCacheDir = envTokenCacheDir
	}
	return Options{
		LoginMethod:   DeviceCodeLogin,
		Environment:   defaultEnvironmentName,
		TokenCacheDir: tokenCacheDir,
	}
}
```

Reply:
@weinong that's how I implemented it my first commit.

The difference is the use of the inline function to avoid adding a new variable, and I find it more elegant.

I pushed a revert